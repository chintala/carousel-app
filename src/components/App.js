import React from 'react';
import AddressFormContainer from './AddressFormContainer';
import CarouselContainer from './CarouselContainer';

class App extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <div className="app">
        <div className="app-wrap">
          <div className="app-header"></div>
          <div className="app-content">
            <AddressFormContainer />
            <CarouselContainer />
          </div>
          <div className="app-footer"></div>
        </div>
      </div>
    );
  }
}

export default App;
