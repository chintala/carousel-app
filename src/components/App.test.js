import React from 'react';
import App from './App';
import {
  render,
  fireEvent,
  cleanup,
  waitForElement
} from '@testing-library/react';
import configureStore from '../store';
import { Provider } from 'react-redux';
import 'jest-dom/extend-expect';

beforeEach(() => {
  fetch.resetMocks();
});

afterEach(cleanup);

it('renders photos', async () => {
  debugger;
  fetch
    .once(
      JSON.stringify({
        results: [{ geometry: { location: { lat: '1', long: '1' } } }]
      })
    )
    .once(
      JSON.stringify({
        photos: {
          photo: [
            { id: '1', farm: 'f', secret: 'sec', server: 's', title: 'title1' },
            { id: '2', farm: 'f', secret: 'sec', server: 's', title: 'title2' },
            { id: '3', farm: 'f', secret: 'sec', server: 's', title: 'title3' }
          ]
        }
      })
    );

  const { container, getByText, getByPlaceholderText, getByAltText } = render(
    <Provider store={configureStore()}>
      <App />
    </Provider>
  );

  const inputNode = getByPlaceholderText('Example: Seattle, WA');
  fireEvent.change(inputNode, {
    target: { value: 'seattle' }
  });
  const buttonNode = getByText('Get Photos');
  fireEvent.click(buttonNode);

  const prevElement = await waitForElement(() => getByText('Previous'));
  expect(prevElement.attributes['disabled']).toBeDefined();

  const nextElement = await waitForElement(() => getByText('Next'));
  expect(nextElement.attributes['disabled']).toBeUndefined();

  const imgElement1 = await waitForElement(() => getByAltText('title1'));
  expect(imgElement1.src).toBe('https://farmf.staticflickr.com/s/1_sec.jpg');

  fireEvent.click(nextElement);

  let imgElement2 = await waitForElement(() => getByAltText('title2'));
  expect(imgElement2.src).toBe('https://farmf.staticflickr.com/s/2_sec.jpg');

  fireEvent.click(nextElement);

  const imgElement3 = await waitForElement(() => getByAltText('title3'));
  expect(imgElement3.src).toBe('https://farmf.staticflickr.com/s/3_sec.jpg');
  expect(nextElement.attributes['disabled']).toBeDefined();

  fireEvent.click(prevElement);

  imgElement2 = await waitForElement(() => getByAltText('title2'));
  expect(imgElement2.src).toBe('https://farmf.staticflickr.com/s/2_sec.jpg');

  expect(container).toMatchSnapshot();
});
