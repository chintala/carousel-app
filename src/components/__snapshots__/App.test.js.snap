// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders photos 1`] = `
<div>
  <div
    class="app"
  >
    <div
      class="app-wrap"
    >
      <div
        class="app-header"
      />
      <div
        class="app-content"
      >
        <form
          class="location-form"
        >
          <input
            placeholder="Example: Seattle, WA"
            type="text"
          />
          <button
            type="submit"
            value="submit"
          >
            Get Photos
          </button>
        </form>
        <div
          class="carousel-container"
        >
          <div
            class="carousel-slide"
          >
            <div
              class="carousel-buttons"
            >
              <button
                type="button"
              >
                Previous
              </button>
              <button
                type="button"
              >
                Play
              </button>
              <button
                type="button"
              >
                 
                Next
              </button>
            </div>
            <div>
              <img
                alt="title2"
                class="carousel-image"
                src="https://farmf.staticflickr.com/s/2_sec.jpg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="app-footer"
      />
    </div>
  </div>
</div>
`;
